src/fenicsxconcrete/gaussian_random_field.py

```diff
@@ -0,0 +1,401 @@
+            cov_name: name of covariance function to use
+            mean: mean of random field
+            rho1: correlation length x
+            rho2: correlation length z
+            sigma2: sigma**2 in covariance function
+            V: FE fct space for problem
+            k: number of eigenvectors to compute
+            ktol: tolerance to chose k (norm eigenvalue_k < ktol)
+            C: correlation matrix
+            M: mass matrix for eigenvalue problem
+            lambdas: eigenvalues
+            EV: eigenvectors
+            field: a representation of the field
+            values: values of the variables (set by user or choosen randomly)
+            values_means: values of the variables for asymptotic expansion to be set by user
+            _type: special case for correlation matrix
+
+        Args:
+            fct_space: FE function space
+            cov_name: string with name of covariance function (e.g. exp, squared_exp ..)
+            mean: mean value of random field
+            rho1: correlation length x
+            rho2: correlation length z
+            sigma: covariance standard deviation
+            k: number of eigenvectors to compute
+            ktol: tolerance to chose k (norm eigenvalue_k < ktol)
+            _type: special cases
+        """
+
+        self.logger = logging.getLogger(__name__)
+
+        self.cov_name = cov_name  # name of covariance function to use
+        self.cov = getattr(self, "cov_" + cov_name)  # select the right function
+        self.mean = mean  # mean of random field
+        # self.rho = rho  # correlation length
+        self.rho1 = rho1  # correlation length x
+        self.rho2 = rho2  # correlation length z
+        self.sigma2 = sigma**2  # sigma**2 in covariance function
+        self.V = fct_space  # FE fct space for problem
+        if k:
+            self.k = k
+        self.ktol = ktol
+
+        self.C = None  # correlation matrix
+        self.M = None  # mass matrix for eigenvalue problem
+        self.lambdas = []  # eigenvalues
+        self.EV = []  # eigenvectors
+        self.k = k  # number of eigenvalues
+        self._type = _type  # specail case for correlation matrix
+
+        self.field = None  # a representation of the field
+        self.values = np.zeros(self.k)  # values of the variables (set by user or choosen randomly)
+
+        self.values_means = np.zeros(self.k)  # values of the variables for asymptotic expansion to be set by user!!
+
+    def __str__(self):
+        """
+        Overloaded description of the field.
+        """
+        name = self.__class__.__name__
+        name += "random field with cov fct %s, mean %s, rho1 %s, rho2 %s, k %s, sig2 %s"
+        return name % (self.cov_name, self.mean, self.rho1, self.rho2, self.k, self.sigma2)
+
+    def __repr__(self):
+        """
+        Overloaded description of the field.
+        """
+
+        return str(self)
+
+    def cov_exp(self, r):
+        """
```

Comment:
Would it be more convenient to use scikit-learn native kernels instead of implementing ours?